From 3dc41265f71a3177c7e6b88d066669dade99386f Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:37 -0400
Subject: [fs] jbd2: don't unplug after writing revoke records

Message-id: <1414058125-4183-124-git-send-email-lczerner@redhat.com>
Patchwork-id: 98240
O-Subject: [RHEL 7.1 PATCH v2 123/171] jbd2: don't unplug after writing revoke records
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit df3c1e9a05ff25aca9f54a6c08b77003e2e32bf1

 During commit process, keep the block device plugged after we are done
 writing the revoke records, until we are finished writing the rest of
 the commit records in the journal.  This will allow most of the
 journal blocks to be written in a single I/O operation, instead of
 separating the the revoke blocks from the rest of the journal blocks.

 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/jbd2/commit.c b/fs/jbd2/commit.c
index cf2fc05..765b31d 100644
--- a/fs/jbd2/commit.c
+++ b/fs/jbd2/commit.c
@@ -555,7 +555,6 @@ void jbd2_journal_commit_transaction(journal_t *journal)
  blk_start_plug(&plug);
  jbd2_journal_write_revoke_records(journal, commit_transaction,
        &log_bufs, WRITE_SYNC);
- blk_finish_plug(&plug);
 
  jbd_debug(3, "JBD2: commit phase 2b\n");
 
@@ -582,7 +581,6 @@ void jbd2_journal_commit_transaction(journal_t *journal)
  err = 0;
  bufs = 0;
  descriptor = NULL;
- blk_start_plug(&plug);
  while (commit_transaction->t_buffers) {
 
   /* Find the next buffer to be journaled... */
-- 
1.7.1